### PR TITLE
Migrated Running Rhino server code

### PR DESCRIPTION
**Pull request by niculistana**
Migrated Running Rhino server code from https://github.com/worldofbal…ance/running-rhino-server to this repository. Still needs testing and debugging.

@jensvdh please check if I did the migration correctly and if follows our naming conventions. If it's okay feel free to merge the pull request so I can delete the branch. I simply copied the same naming convention as the other packages like what you talked about in class.

### CODE REVIEW

File: src/rr/core/GameManager.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+///*
```
Comment (jensvdh) [inline src/rr/core/GameManager.java:1]: Will you ever need this file?

If we are just completely commenting it out we can also delete it. Git will keep a history incase you ever want it back regardless.

**Comment by jensvdh** [comment]
I don't see a Main Method anywhere? (Except in testDB connection). How would we compile this into a JAR and run this as its own minigameserver?

**Comment by niculistana** [comment]
@jensvdh I have not fully migrated the `rr.core` package yet. I wanted to clarify if you wanted to subclass the RR GameClient and GameServer from the shared.core packages. It seems to me that we are keeping separate from the shared.core packages, which I think is the way to go.

 The main method is supposed to be under here: https://github.com/worldofbalance/running-rhino-server/tree/development/src/core

**Comment by jensvdh** [comment]
Try to reuse code from shared as much as possible. If you think parts of your code can be used by other teams, then put them under shared.

If that makes sense?

File: src/rr/core/GameManager.java (hunk 1 above)
Comment (niculistana) [inline src/rr/core/GameManager.java:1]: Will finish uploading the rest of rr.core later tonight.

EDIT: I just finished. There are some files that are all commented out. I am still investigating if they are okay to remove. I believe the last semester's team wanted to do something with them but could not have time to finish implementing.

**Comment by niculistana** [comment]
@jensvdh Yes, for now I am only using logger and parser codes from shared since they seem generic to every game. I am still looking around to see which ones we can use from shared. I am inspecting `rr.util` and `rr.metadata` for more duplicate items. Thank you.

**Comment by jensvdh** [comment]
Would this PR also take care of #3 ?

**Comment by jensvdh** [comment]
- [ ]

**Comment by niculistana** [comment]
@jensvdh no, this is simply refactoring. I have not fixed #3 yet.

**Comment by niculistana** [comment]
@haichuand Feel free to commit to this branch.

**Comment by jensvdh** [comment]
Let me know when this PR is ready for review.

**Comment by niculistana** [comment]
It is. Also, @haichuand pushed his fix for #3 over here.

**Comment by niculistana** [comment]
@jensvdh Looks like we can merge the PR. Lmk if its ok.

**Comment by jensvdh** [comment]
Hey, this is quite the monster of a pull request. I'll skim through it later tonight.

**Comment by niculistana** [comment]
Haha, sounds good. I'll keep my eyes peeled for notifications. -nicu

On Fri, Apr 1, 2016 at 2:42 PM, Jens Vanderhaeghe notifications@github.com
wrote:

> Hey, this is quite the monster of a pull request. I'll skim through it
> later tonight.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> https://github.com/worldofbalance/wob-server/pull/4#issuecomment-204575552

## 

Nicu Listana
_LinkedIn
http://www.linkedin.com/profile/view?id=134425825&locale=en_US&trk=tab_pro.Twitter
https://twitter.com/#!/apebeast.G+
https://plus.google.com/b/101191476444716618590/101191476444716618590/_

**Comment by anuagg** [comment]
Are you guys able to play running rhino through this code? I only get to a point where both the players can choose their player specie. Though there are no exceptions now when we run the server code, I think we don't really have the minigame servers started at the given port. Jar files get executed to a point where run() method is called inside main (for GameServer class inside jar). But I think the run function is not getting executed [ it says run() function not used - See attached pic] . Is it meant to behave this way ? If its running fine for everyone else then it might be a windows issue or something.
capture

**Comment by niculistana** [comment]
@anuagg, is your client running the server locally or the hosted server?

**Comment by anuagg** [comment]
@niculistana  Locally

**Comment by niculistana** [comment]
@anuagg please see my answer at #11. I ran my server locally, and it is working.

**Pull request merged**